external/js/cothority/lib/identity.js

```diff
@@ -0,0 +1,227 @@
+     * @return a ServerIdentity
+     * */
+    static fromHexPublic(hexPublic,address,description) {
+        const pub = misc.hexToUint8Array(hexPublic);
+        return new ServerIdentity(pub,address,description);
+    }
+    /*
+     * @return the public key as a Uint8Array buffer
+     * */
+    get public() {
+        return this.pub;
+    }
+
+    /*
+     * @return the WebSocket address. That can be passed into the net.Socket
+     * address constructor argument.
+     * */
+    get websocketAddr() {
+        return this.wsAddr;
+    }
+
+    /*
+     * @return the tcp address of the server
+     * */
+    get tcpAddr() {
+        return this.addr;
+    }
+
+    /*
+     * @return the id of this serveridentity
+     * */
+    get id() {
+        return this._id;
+    }
+
+    /**
+     * returns the description associated with this identity
+     *
+     * @returns {string} the string description
+     */
+    get description() {
+        return this._description;
+    }
+
+    toString() {
+        return this.tcpAddr;
+    }
+
+    /**
+     * point returns the point representation of the public key
+     *
+     * @param {kyber.Group} group the expected group from which the public key belongs
+     * @returns {kyber.Point} the kyber.Point representing the public key
+     */
+    point(group) {
+    /*    if (this.point)*/
+            //return this.point;
+
+        //this.point = group.point();
+        //this.point.unmarshalBinary(this.pub);
+        /*return this.point.clone();*/
+        const point = group.point();
+        point.unmarshalBinary(this.pub);
+        return point;
+    }
+}
+
+
+/*
+ * Roster represents a group of servers. It basically consists in a list of
+ * ServerIdentity with some helper functions.
+ * */
+class Roster {
+
+    /*
+     * @param a list of ServerIdentity
+     * @return a Roster from the given list of identites
+     * */
+    constructor(identities,id) {
+        this._identities = identities;
+        this._id = id;
+    }
+
+    /*
+     * Random selects a random identity and returns it
+     * @return a random identity
+     * */
+    random() {
+        const idx = Math.floor(Math.random() * (this.length()-1));
+        return this._identities[idx];
+    }
+
+    /*
+     * @return the list of identities composing this Roster
+     * */
+    get identities() {
+        return this._identities;
+    }
+
+    /*
+     * @return the length of the roster
+     * */
+    get length() {
+        return this._identities.length;
+    }
+
+    /*
+     * @return the id of the roster
+     * */
+    get id() {
+        return this._id;
+    }
+
+    /**
+     * aggregateKey returns the aggregate public key for this server.
+     * It is the sum of all public keys of the identities of this Roster.
+     *
+     * @param {kyber.Group} group The group to use to compute the aggregate key.
+     * @returns {kyber.Point} The aggregate key
+     */
+    aggregateKey(group) {
```

Comment:
As a continuation of the previous comment: if the point was stored in the identity, you'd already know the group and not need to pass it in here.